lib/i18n/js/engine.rb

```diff
@@ -9,56 +10,58 @@ class SprocketsExtension
     end
 
     class Engine < ::Rails::Engine
-      if JS::Dependencies.sprockets_supports_register_preprocessor?
```

Comment:
I guess you need to require `sprockets/version` with `rescue LoadError`?
Again sprockets might be absent ~_~